### PR TITLE
Make agency and route optional for ptFares

### LINKED ISSUE #1048: Make agency and route optional for ptFares

### PR DESCRIPTION
**Pull request by zishanbilal**
Fixed #1048

---
This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/1052)

### CODE REVIEW

File: src/main/scala/beam/agentsim/agents/choice/mode/PtFares.scala [hunk 1]
```diff
@@ -4,31 +4,74 @@ import java.io.FileNotFoundException
...
-      .getOrElse(agencyId, List())
+  def getPtFare(agencyId: Option[String], routeId: Option[String], age: Option[Int]): Option[Double] = {
+    val ageFilter = {
+      val ageMatch = ptFares
+        .filter(
+          f =>
+            age.fold(false)(f.age.has)
+        )
+
+      lazy val ageNoMatch = ptFares
```
Comment (JustinPihony) [inline src/main/scala/beam/agentsim/agents/choice/mode/PtFares.scala]: It took a bit of thinking to discern this. However I'm still not sure the naming and the end result match up to me. What is this supposed to yield? I'm wondering if a diff using `ageMatch` would be more easily understood to a reader of the code? Or I missed something..

Comment (JustinPihony) [reply src/main/scala/beam/agentsim/agents/choice/mode/PtFares.scala]: Admittedly, the whole flow takes a bit to understand...Could this be extracted into a method that has a name of what it is doing ... as a guide for those reading the code?

**Comment by colinsheppard** [comment]
test!

**Comment by zishanbilal** [comment]
> test!

https://github.com/LBNL-UCB-STI/beam/pull/1088

**Comment by colinsheppard** [comment]
test!

**Comment by colinsheppard** [comment]
test!

**Pull request merged**